src/mock/mock_builder.ts

```diff
@@ -99,6 +99,7 @@ export class MockBuilder<ClassToMock> extends TestDoubleBuilder<ClassToMock> {
       value: (...args: unknown[]) => {
         // Track that this method was called
         mock.calls[method]++;
+        // TODO: copy spy approach because we need mock.expected_args
```

Comment:
missed todo? not sure if this is crucial or could be done later down the line

Reply:
i didn't want to bloat the PR more with hella changes so i added these two issues:

https://github.com/drashland/rhum/issues/166
https://github.com/drashland/rhum/issues/167